volmdlr/core_compiled.pyx

```diff
@@ -379,8 +379,15 @@ class Vector2D(Vector):
 
     @classmethod
     def remove_duplicate(cls, points):
-        dict_ = {p.approx_hash() : p for p in points}
-        return list(dict_.values())
+        """
+        Don't keep the input list order
+        """
+        input_points = sorted(points, key=lambda pt: (pt.x, pt.y))
+        new_points = [points[0]]
+        for i, point in enumerate(input_points[1:]):
+            if not point.is_close(input_points[i]):
```

Comment:
`point` should be compared to last point added in `new_points`, not to `input_points[i]`, in case several close points are following each other.

```suggestion
            if not point.is_close(new_points[-1]):
```

Reply:
Done